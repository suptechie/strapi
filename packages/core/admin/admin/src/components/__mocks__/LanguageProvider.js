export function useLocalesProvider() {
  return {
    changeLocale() {},
    localeNames: { en: 'English' },
    messages: ['test'],
  };
}
