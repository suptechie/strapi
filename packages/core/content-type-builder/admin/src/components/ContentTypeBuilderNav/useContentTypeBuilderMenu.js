import { useState } from 'react';
import { useNotification, useTracking } from '@strapi/helper-plugin';
import { camelCase, isEmpty, sortBy, toLower, upperFirst } from 'lodash';
import matchSorter from 'match-sorter';
import { useIntl } from 'react-intl';
import { useHistory } from 'react-router-dom';
import useDataManager from '../../hooks/useDataManager';
import pluginId from '../../pluginId';
import getTrad from '../../utils/getTrad';
import makeSearch from '../../utils/makeSearch';

const useContentTypeBuilderMenu = () => {
  const {
    components,
    componentsGroupedByCategory,
    contentTypes,
    isInDevelopmentMode,
    sortedContentTypesList,
  } = useDataManager();
  const toggleNotification = useNotification();
  const { trackUsage } = useTracking();
  const { formatMessage } = useIntl();
  const { push } = useHistory();
  const [search, setSearch] = useState('');

  const componentsData = sortBy(
    Object.keys(componentsGroupedByCategory).map(category => ({
      name: category,
      title: category,
      isEditable: isInDevelopmentMode,
      onClickEdit: (e, data) => {
        e.stopPropagation();

        const search = makeSearch({
          actionType: 'edit',
          modalType: 'editCategory',
          categoryName: data.name,
          header_label_1: formatMessage({
            id: getTrad('modalForm.header.categories'),
            defaultMessage: 'Categories',
          }),
          header_info_category_1: null,
          header_info_name_1: null,
          header_label_2: data.name,
          header_info_category_2: null,
          header_info_name_2: null,

          settingType: 'base',
        });

        push({ search });
      },
      links: sortBy(
        componentsGroupedByCategory[category].map(compo => ({
          name: compo.uid,
          to: `/plugins/${pluginId}/component-categories/${category}/${compo.uid}`,
          title: compo.schema.name,
        })),
        obj => obj.title
      ),
    })),
    obj => obj.title
  );

  const canOpenModalCreateCTorComponent = () => {
    return (
      !Object.keys(contentTypes).some(ct => contentTypes[ct].isTemporary === true) &&
      !Object.keys(components).some(component => components[component].isTemporary === true)
    );
  };

  const handleClickOpenModal = async (modalType, kind = '') => {
    const type = kind === 'singleType' ? kind : modalType;

    if (canOpenModalCreateCTorComponent()) {
      trackUsage(`willCreate${upperFirst(camelCase(type))}`);

      const search = makeSearch({
        modalType,
        kind,
        actionType: 'create',
        settingType: 'base',
        forTarget: modalType,
        header_label_1: 'null',
      });
      push({
        search,
      });
    } else {
      toggleNotification({
        type: 'info',
        message: {
          id: `${getTrad('notification.info.creating.notSaved')}`,
          defaultMessage:
            'Please save your work before creating a new collection type or component',
        },
      });
    }
  };

  const displayedContentTypes = sortedContentTypesList.filter(obj => obj.visible);

  const data = [
    {
      name: 'models',
      title: {
        id: `${getTrad('menu.section.models.name.')}`,
        defaultMessage: 'Collection Types',
      },
      customLink: isInDevelopmentMode && {
        id: `${getTrad('button.model.create')}`,
        defaultMessage: 'Create new collection type',
        onClick: () => handleClickOpenModal('contentType', 'collectionType'),
      },
      links: displayedContentTypes.filter(contentType => contentType.kind === 'collectionType'),
    },
    {
      name: 'singleTypes',
      title: {
        id: `${getTrad('menu.section.single-types.name.')}`,
        defaultMessage: 'Single Types',
      },
      customLink: isInDevelopmentMode && {
        id: `${getTrad('button.single-types.create')}`,
        defaultMessage: 'Create new single type',
        onClick: () => handleClickOpenModal('contentType', 'singleType'),
      },
      links: displayedContentTypes.filter(singleType => singleType.kind === 'singleType'),
    },
    {
      name: 'components',
      title: {
        id: `${getTrad('menu.section.components.name.')}`,
        defaultMessage: 'Components',
      },
      customLink: {
        id: `${getTrad('button.component.create')}`,
        defaultMessage: 'Create a new component',
        onClick: () => handleClickOpenModal('component'),
      },
      links: componentsData,
    },
  ];

  const matchByTitle = links =>
    matchSorter(links, toLower(search), { keys: [item => toLower(item.title)] });

  const getMenu = () => {
    // Maybe we can do it simpler with matchsorter wildcards ?
    return data.map(section => {
      const hasChild = section.links.some(l => !isEmpty(l.links));

      if (hasChild) {
        return {
          ...section,
          links: section.links.map(l => ({ ...l, links: matchByTitle(l.links) })),
        };
      }

      return {
        ...section,
        links: matchByTitle(section.links),
      };
    });
  };

  return { menu: getMenu(), searchValue: search, onSearchChange: setSearch };
};

export default useContentTypeBuilderMenu;
