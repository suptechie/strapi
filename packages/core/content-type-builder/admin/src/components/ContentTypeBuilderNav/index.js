import {
  SubNav,
  SubNavHeader,
  SubNavLink,
  SubNavLinkSection,
  SubNavSection,
  SubNavSections,
  Box,
} from '@strapi/parts';
import { TextButton } from '@strapi/parts/TextButton';
import { AddIcon } from '@strapi/icons';
import upperFirst from 'lodash/upperFirst';
import React from 'react';
import { useIntl } from 'react-intl';
import useContentTypeBuilderMenu from './useContentTypeBuilderMenu';
import pluginId from '../../pluginId';

// ! ASK SOUP ABOUT THE WAIT + TOGGLEPROMPT
const wait = () => new Promise(resolve => setTimeout(resolve, 100));

const ContentTypeBuilderNav = () => {
  const { menu, searchValue, onSearchChange } = useContentTypeBuilderMenu({ wait });
  const { formatMessage } = useIntl();

  return (
    <SubNav ariaLabel={`${pluginId}.plugin.name`}>
      <SubNavHeader
        searchable
        value={searchValue}
        onClear={() => onSearchChange('')}
        onChange={e => onSearchChange(e.target.value)}
        label={formatMessage({ id: `${pluginId}.plugin.name` })}
        searchLabel="Search..."
      />
      <SubNavSections>
        {menu.map(section => {
          const title = `${section.title.id}${section.links.length > 1 ? 'plural' : 'singular'}`;

          return (
            <>
              <SubNavSection
                key={section.name}
                label={formatMessage({ id: title, defaultMessage: title.defaultMessage })}
                collapsable
                badgeLabel={section.links.length.toString()}
              >
                {section.links.map(link => {
                  if (link.links) {
                    return (
                      <SubNavLinkSection key={link.title} label={upperFirst(link.title)}>
                        {link.links.map(subLink => (
                          <SubNavLink to={subLink.to} active={subLink.active} key={subLink.uid}>
                            {upperFirst(
                              formatMessage({ id: subLink.name, defaultMessage: subLink.title })
                            )}
                          </SubNavLink>
                        ))}
                      </SubNavLinkSection>
                    );
                  }

                  return (
                    <SubNavLink to={link.to} active={link.active} key={link.uid}>
                      {upperFirst(formatMessage({ id: link.name, defaultMessage: link.title }))}
                    </SubNavLink>
                  );
                })}
              </SubNavSection>
              {section.customLink && (
                <Box as="li" paddingLeft={7}>
                  <TextButton onClick={section.customLink.onClick} startIcon={<AddIcon />}>
                    {formatMessage({
                      id: section.customLink.id,
                      defaultMessage: section.customLink.id,
                    })}
                  </TextButton>
                </Box>
              )}
            </>
          );
        })}
      </SubNavSections>
    </SubNav>
  );
};

export default ContentTypeBuilderNav;
