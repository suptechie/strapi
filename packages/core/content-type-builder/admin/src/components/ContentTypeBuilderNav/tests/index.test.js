/**
 *
 * Tests for ContentTypeBuilderNav
 *
 */

import { Layout } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import LanguageProvider from '../../../../../../admin/admin/src/components/LanguageProvider';
import Theme from '../../../../../../admin/admin/src/components/Theme';
import en from '../../../../../../admin/admin/src/translations/en.json';
import ContentTypeBuilderNav from '../index';
import mockData from './mockData';

jest.mock('../useContentTypeBuilderMenu.js', () => {
  return jest.fn(() => ({
    menu: mockData,
    searchValue: '',
    onSearchChange: () => {},
  }));
});

const makeApp = () => {
  const history = createMemoryHistory();
  const messages = { en };
  const localeNames = { en: 'English' };

  return (
    <LanguageProvider messages={messages} localeNames={localeNames}>
      <Theme>
        <Router history={history}>
          <Layout sideNav={<ContentTypeBuilderNav />}>
            <div />
          </Layout>
        </Router>
      </Theme>
    </LanguageProvider>
  );
};

describe('<ContentTypeBuilderNav />', () => {
  it('renders and matches the snapshot', () => {
    const App = makeApp(ContentTypeBuilderNav);
    const { container } = render(App);

    expect(container).toMatchSnapshot();
  });
});
