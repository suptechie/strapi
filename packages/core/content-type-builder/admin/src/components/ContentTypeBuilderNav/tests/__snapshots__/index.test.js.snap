// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentTypeBuilderNav /> renders and matches the snapshot 1`] = `
.c40 {
  padding-bottom: 56px;
}

.c0 {
  display: grid;
  grid-template-columns: auto 1fr;
}

.c41 {
  overflow-x: hidden;
}

.c42 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c30 {
  padding-left: 32px;
}

.c2 {
  padding-top: 24px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-left: 24px;
}

.c8 {
  padding-top: 16px;
}

.c9 {
  background: #eaeaef;
}

.c12 {
  padding-top: 8px;
  padding-bottom: 16px;
}

.c14 {
  padding-top: 8px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-left: 24px;
}

.c19 {
  padding-right: 4px;
}

.c22 {
  background: #eaeaef;
  padding: 4px;
  border-radius: 4px;
}

.c25 {
  background: #f6f6f9;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 32px;
}

.c28 {
  padding-left: 8px;
}

.c35 {
  padding-top: 8px;
  padding-right: 16px;
  padding-bottom: 8px;
  padding-left: 32px;
}

.c1 {
  width: 14.5rem;
  background: #f6f6f9;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  height: 100vh;
  overflow-y: auto;
  border-right: 1px solid #dcdce4;
  z-index: 1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  color: #32324d;
  font-weight: 600;
  font-size: 1.125rem;
  line-height: 1.22;
}

.c20 {
  color: #666687;
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
}

.c29 {
  color: #32324d;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c39 {
  font-weight: 500;
  color: #32324d;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c6 svg {
  height: 12px;
  width: 12px;
}

.c6 svg > g,
.c6 svg path {
  fill: #ffffff;
}

.c6[aria-disabled='true'] {
  pointer-events: none;
}

.c6:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c6:focus-visible {
  outline: none;
}

.c6:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  width: 2rem;
}

.c7 svg > g,
.c7 svg path {
  fill: #8e8ea9;
}

.c7:hover svg > g,
.c7:hover svg path {
  fill: #666687;
}

.c7:active svg > g,
.c7:active svg path {
  fill: #a5a5ba;
}

.c7[aria-disabled='true'] {
  background-color: #eaeaef;
}

.c7[aria-disabled='true'] svg path {
  fill: #666687;
}

.c10 {
  height: 1px;
  border: none;
  margin: 0;
}

.c11 {
  width: 1.5rem;
  background-color: #dcdce4;
}

.c26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #32324d;
}

.c26 svg > * {
  fill: #666687;
}

.c26.active {
  background-color: #f0f0ff;
  border-right: 2px solid #4945ff;
}

.c26.active svg > * {
  fill: #271fe0;
}

.c26.active .c4 {
  color: #271fe0;
  font-weight: 500;
}

.c26:focus-visible {
  outline-offset: -2px;
}

.c27 {
  width: 0.75rem;
  height: 0.25rem;
}

.c27 * {
  fill: #666687;
}

.c36 {
  max-height: 2rem;
}

.c36 svg {
  height: 0.25rem;
}

.c36 svg path {
  fill: #4a4a6a;
}

.c37 {
  border: none;
  padding: 0;
  background: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c38 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 0.75rem;
  -webkit-transform: rotateX(0deg);
  -ms-transform: rotateX(0deg);
  transform: rotateX(0deg);
}

.c23 {
  display: inline-block;
}

.c18 {
  border: none;
  padding: 0;
  background: transparent;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: rotateX(0deg);
  -ms-transform: rotateX(0deg);
  transform: rotateX(0deg);
}

.c15 {
  max-height: 2rem;
}

.c15 svg {
  height: 0.25rem;
}

.c15 svg path {
  fill: #8e8ea9;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c13 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c13 > * + * {
  margin-top: 8px;
}

.c33 {
  padding-right: 8px;
}

.c34 {
  color: #4945ff;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c32 {
  background: transparent;
  border: none;
  position: relative;
  outline: none;
}

.c32[aria-disabled='true'] {
  pointer-events: none;
}

.c32[aria-disabled='true'] svg path {
  fill: #666687;
}

.c32 svg {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.625rem;
}

.c32 svg path {
  fill: #4945ff;
}

.c32:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c32:focus-visible {
  outline: none;
}

.c32:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

<div>
  <div
    class="c0"
  >
    <nav
      aria-label="Content-Types Builder"
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Content-Types Builder
          </h2>
          <span>
            <button
              aria-disabled="false"
              aria-labelledby="tooltip-2"
              class="c6 c7"
              tabindex="0"
              type="button"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M23.813 20.163l-5.3-5.367a9.792 9.792 0 001.312-4.867C19.825 4.455 15.375 0 9.913 0 4.45 0 0 4.455 0 9.929c0 5.473 4.45 9.928 9.912 9.928a9.757 9.757 0 005.007-1.4l5.275 5.35a.634.634 0 00.913 0l2.706-2.737a.641.641 0 000-.907zM9.91 3.867c3.338 0 6.05 2.718 6.05 6.061s-2.712 6.061-6.05 6.061c-3.337 0-6.05-2.718-6.05-6.06 0-3.344 2.713-6.062 6.05-6.062z"
                  fill="#32324D"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </span>
        </div>
        <div
          class="c8"
        >
          <hr
            class="c9 c10 c11"
          />
        </div>
      </div>
      <div
        class="c12"
      >
        <ul
          class="c13"
        >
          <li>
            <div
              class="c14 c15"
            >
              <div
                class="c16"
              >
                <button
                  aria-controls="subnav-list-4"
                  aria-expanded="true"
                  class="c17 c18"
                >
                  <div
                    class="c19"
                  >
                    <span
                      class="c4 c20"
                    >
                      content-type-builder.menu.section.models.name.
                    </span>
                  </div>
                  <div
                    class="c21"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      height="1em"
                      viewBox="0 0 14 8"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                        fill="#32324D"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                </button>
                <div
                  class="c22 c23 c24"
                >
                  <span
                    class="c4 c20"
                  >
                    2
                  </span>
                </div>
              </div>
            </div>
            <ul
              id="subnav-list-4"
            >
              <li>
                <a
                  class="c25 c26"
                  href="/plugins/content-type-builder/content-types/application::address.address"
                >
                  <div
                    class="c17"
                  >
                    <svg
                      class="c27"
                      fill="none"
                      height="1em"
                      viewBox="0 0 4 4"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="#A5A5BA"
                        height="4"
                        rx="2"
                        width="4"
                      />
                    </svg>
                    <div
                      class="c28"
                    >
                      <span
                        class="c4 c29"
                      >
                        Address
                      </span>
                    </div>
                  </div>
                </a>
              </li>
              <li>
                <a
                  class="c25 c26"
                  href="/plugins/content-type-builder/content-types/application::category.category"
                >
                  <div
                    class="c17"
                  >
                    <svg
                      class="c27"
                      fill="none"
                      height="1em"
                      viewBox="0 0 4 4"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="#A5A5BA"
                        height="4"
                        rx="2"
                        width="4"
                      />
                    </svg>
                    <div
                      class="c28"
                    >
                      <span
                        class="c4 c29"
                      >
                        Category
                      </span>
                    </div>
                  </div>
                </a>
              </li>
            </ul>
          </li>
          <li
            class="c30"
          >
            <button
              aria-disabled="false"
              class="c31 c32"
              type="button"
            >
              <span
                aria-hidden="true"
                class="c33"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 13.604a.3.3 0 01-.3.3h-9.795V23.7a.3.3 0 01-.3.3h-3.21a.3.3 0 01-.3-.3v-9.795H.3a.3.3 0 01-.3-.3v-3.21a.3.3 0 01.3-.3h9.795V.3a.3.3 0 01.3-.3h3.21a.3.3 0 01.3.3v9.795H23.7a.3.3 0 01.3.3v3.21z"
                    fill="#212134"
                  />
                </svg>
              </span>
              <span
                class="c34"
              >
                Create new collection type
              </span>
            </button>
          </li>
          <li>
            <div
              class="c14 c15"
            >
              <div
                class="c16"
              >
                <button
                  aria-controls="subnav-list-5"
                  aria-expanded="true"
                  class="c17 c18"
                >
                  <div
                    class="c19"
                  >
                    <span
                      class="c4 c20"
                    >
                      content-type-builder.menu.section.single-types.name.
                    </span>
                  </div>
                  <div
                    class="c21"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      height="1em"
                      viewBox="0 0 14 8"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                        fill="#32324D"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                </button>
                <div
                  class="c22 c23 c24"
                >
                  <span
                    class="c4 c20"
                  >
                    1
                  </span>
                </div>
              </div>
            </div>
            <ul
              id="subnav-list-5"
            >
              <li>
                <a
                  class="c25 c26"
                  href="/plugins/content-type-builder/content-types/application::homepage.homepage"
                >
                  <div
                    class="c17"
                  >
                    <svg
                      class="c27"
                      fill="none"
                      height="1em"
                      viewBox="0 0 4 4"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="#A5A5BA"
                        height="4"
                        rx="2"
                        width="4"
                      />
                    </svg>
                    <div
                      class="c28"
                    >
                      <span
                        class="c4 c29"
                      >
                        Homepage
                      </span>
                    </div>
                  </div>
                </a>
              </li>
            </ul>
          </li>
          <li
            class="c30"
          >
            <button
              aria-disabled="false"
              class="c31 c32"
              type="button"
            >
              <span
                aria-hidden="true"
                class="c33"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 13.604a.3.3 0 01-.3.3h-9.795V23.7a.3.3 0 01-.3.3h-3.21a.3.3 0 01-.3-.3v-9.795H.3a.3.3 0 01-.3-.3v-3.21a.3.3 0 01.3-.3h9.795V.3a.3.3 0 01.3-.3h3.21a.3.3 0 01.3.3v9.795H23.7a.3.3 0 01.3.3v3.21z"
                    fill="#212134"
                  />
                </svg>
              </span>
              <span
                class="c34"
              >
                Create new single type
              </span>
            </button>
          </li>
          <li>
            <div
              class="c14 c15"
            >
              <div
                class="c16"
              >
                <button
                  aria-controls="subnav-list-6"
                  aria-expanded="true"
                  class="c17 c18"
                >
                  <div
                    class="c19"
                  >
                    <span
                      class="c4 c20"
                    >
                      content-type-builder.menu.section.components.name.
                    </span>
                  </div>
                  <div
                    class="c21"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      height="1em"
                      viewBox="0 0 14 8"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                        fill="#32324D"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                </button>
                <div
                  class="c22 c23 c24"
                >
                  <span
                    class="c4 c20"
                  >
                    2
                  </span>
                </div>
              </div>
            </div>
            <ul
              id="subnav-list-6"
            >
              <li>
                <div
                  class="c35 c36"
                >
                  <div
                    class="c16"
                  >
                    <button
                      aria-controls="subnav-list-7"
                      aria-expanded="true"
                      class="c37"
                    >
                      <div
                        class="c38"
                      >
                        <svg
                          aria-hidden="true"
                          fill="none"
                          height="1em"
                          viewBox="0 0 14 8"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                            fill="#32324D"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </div>
                      <div
                        class="c28"
                      >
                        <span
                          class="c4 c39"
                        >
                          Basic
                        </span>
                      </div>
                    </button>
                  </div>
                </div>
                <ul
                  id="subnav-list-7"
                >
                  <li>
                    <a
                      class="c25 c26"
                      href="/plugins/content-type-builder/component-categories/basic/basic.simple"
                    >
                      <div
                        class="c17"
                      >
                        <svg
                          class="c27"
                          fill="none"
                          height="1em"
                          viewBox="0 0 4 4"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="#A5A5BA"
                            height="4"
                            rx="2"
                            width="4"
                          />
                        </svg>
                        <div
                          class="c28"
                        >
                          <span
                            class="c4 c29"
                          >
                            Simple
                          </span>
                        </div>
                      </div>
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <div
                  class="c35 c36"
                >
                  <div
                    class="c16"
                  >
                    <button
                      aria-controls="subnav-list-8"
                      aria-expanded="true"
                      class="c37"
                    >
                      <div
                        class="c38"
                      >
                        <svg
                          aria-hidden="true"
                          fill="none"
                          height="1em"
                          viewBox="0 0 14 8"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M14 .889a.86.86 0 01-.26.625L7.615 7.736A.834.834 0 017 8a.834.834 0 01-.615-.264L.26 1.514A.861.861 0 010 .889c0-.24.087-.45.26-.625A.834.834 0 01.875 0h12.25c.237 0 .442.088.615.264a.86.86 0 01.26.625z"
                            fill="#32324D"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </div>
                      <div
                        class="c28"
                      >
                        <span
                          class="c4 c39"
                        >
                          Default
                        </span>
                      </div>
                    </button>
                  </div>
                </div>
                <ul
                  id="subnav-list-8"
                >
                  <li>
                    <a
                      class="c25 c26"
                      href="/plugins/content-type-builder/component-categories/default/default.closingperiod"
                    >
                      <div
                        class="c17"
                      >
                        <svg
                          class="c27"
                          fill="none"
                          height="1em"
                          viewBox="0 0 4 4"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="#A5A5BA"
                            height="4"
                            rx="2"
                            width="4"
                          />
                        </svg>
                        <div
                          class="c28"
                        >
                          <span
                            class="c4 c29"
                          >
                            Closingperiod
                          </span>
                        </div>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      class="c25 c26"
                      href="/plugins/content-type-builder/component-categories/default/default.dish"
                    >
                      <div
                        class="c17"
                      >
                        <svg
                          class="c27"
                          fill="none"
                          height="1em"
                          viewBox="0 0 4 4"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="#A5A5BA"
                            height="4"
                            rx="2"
                            width="4"
                          />
                        </svg>
                        <div
                          class="c28"
                        >
                          <span
                            class="c4 c29"
                          >
                            Dish
                          </span>
                        </div>
                      </div>
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li
            class="c30"
          >
            <button
              aria-disabled="false"
              class="c31 c32"
              type="button"
            >
              <span
                aria-hidden="true"
                class="c33"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 13.604a.3.3 0 01-.3.3h-9.795V23.7a.3.3 0 01-.3.3h-3.21a.3.3 0 01-.3-.3v-9.795H.3a.3.3 0 01-.3-.3v-3.21a.3.3 0 01.3-.3h9.795V.3a.3.3 0 01.3-.3h3.21a.3.3 0 01.3.3v9.795H23.7a.3.3 0 01.3.3v3.21z"
                    fill="#212134"
                  />
                </svg>
              </span>
              <span
                class="c34"
              >
                Create a new component
              </span>
            </button>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="c40 c41"
    >
      <div />
    </div>
  </div>
  <div
    class="c42"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
